homeassistant/components/lektrico/switch.py

```diff
@@ -0,0 +1,116 @@
+"""Support for Lektrico switch entities."""
+
+from collections.abc import Callable, Coroutine
+from dataclasses import dataclass
+from typing import Any
+
+from lektricowifi import Device
+
+from homeassistant.components.switch import SwitchEntity, SwitchEntityDescription
+from homeassistant.const import ATTR_SERIAL_NUMBER, CONF_TYPE, EntityCategory
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import LektricoConfigEntry, LektricoDeviceDataUpdateCoordinator
+from .entity import LektricoEntity
+
+
+@dataclass(frozen=True, kw_only=True)
+class LektricoSwitchEntityDescription(SwitchEntityDescription):
+    """Describes Lektrico switch entity."""
+
+    value_fn: Callable[[dict[str, Any]], bool]
+    set_value_fn: Callable[[Device, dict[Any, Any], bool], Coroutine[Any, Any, Any]]
+
+
+SWITCHS_FOR_ALL_CHARGERS: tuple[LektricoSwitchEntityDescription, ...] = (
+    LektricoSwitchEntityDescription(
+        key="authentication",
+        translation_key="authentication",
+        entity_category=EntityCategory.CONFIG,
+        value_fn=lambda data: bool(data["require_auth"]),
+        set_value_fn=lambda device, data, value: device.set_auth(not value),
+    ),
+    LektricoSwitchEntityDescription(
+        key="lock",
+        translation_key="lock",
+        entity_category=EntityCategory.CONFIG,
+        value_fn=lambda data: str(data["charger_state"]) == "locked",
+        set_value_fn=lambda device, data, value: device.set_charger_locked(value),
+    ),
+)
+
+
+SWITCHS_FOR_3_PHASE_CHARGERS: tuple[LektricoSwitchEntityDescription, ...] = (
+    LektricoSwitchEntityDescription(
+        key="force_single_phase",
+        translation_key="force_single_phase",
+        entity_category=EntityCategory.CONFIG,
+        value_fn=lambda data: data["relay_mode"] == 1,
+        set_value_fn=lambda device, data, value: device.set_relay_mode(
+            data["dynamic_current"], 1
+        )
+        if value
+        else device.set_relay_mode(data["dynamic_current"], 3),
```

Comment:
Please wrap multiline lambdas in parenthesis

Reply:
I added the parenthesis, thank you.